core/DataPrefetcher.cpp

```diff
@@ -0,0 +1,121 @@
+#include "DataPrefetcher.hpp"
+#include "NextLinePrefetchEngine.hpp"
+
+namespace olympia
+{
+
+    DataPrefetcher::DataPrefetcher(sparta::TreeNode* node, const DataPrefetcherParameterSet* p) :
+        sparta::Unit(node),
+        PrefetcherIF<PrefetchEngineIF<>>(dynamic_cast<sparta::Unit*>(this)),
```

Comment:
Is there a way to avoid the dynamic cast? I think you can use the `getResourceAs` Sparta method like this:
```
node->getResourceAs<sparta::Unit*>
```

Reply:
I will remove the dynamic cast